libr/anal/type.c

```diff
@@ -122,18 +122,108 @@ R_API void r_anal_save_parsed_type(RAnal *anal, const char *parsed) {
 static int typecmp(const void *a, const void *b) {
 	char *type1 = (char *) a;
 	char *type2 = (char *) b;
-	return strcmp (type1 , type2);
+	return strcmp (type1, type2);
 }
 
 R_API RList *r_anal_types_from_fcn(RAnal *anal, RAnalFunction *fcn) {
 	RListIter *iter;
 	RAnalVar *var;
 	RList *list = r_anal_var_all_list (anal, fcn);
-        RList *type_used = r_list_new ();
-        r_list_foreach (list, iter, var) {
-		r_list_append (type_used , var->type);
-        }
-	RList *uniq = r_list_uniq (type_used , typecmp);
+	RList *type_used = r_list_new ();
+	r_list_foreach (list, iter, var) {
+		r_list_append (type_used, var->type);
+	}
+	RList *uniq = r_list_uniq (type_used, typecmp);
 	r_list_free (type_used);
 	return uniq;
 }
+
+static void enum_type_fini(void *e, void *user) {
+	(void)user;
+	RAnalEnumCase *cas = e;
+	free ((char *)cas->name);
+}
+
+static RAnalBaseType *get_enum_type(RAnal *anal, const char *sanitized_name) {
+	assert (sanitized_name != NULL);
+
+	RAnalBaseType *base_type = malloc (sizeof(RAnalBaseType));
+	if (!base_type) {
+		return NULL;
+	}
+	base_type->kind = R_ANAL_BASE_TYPE_KIND_ENUM;
+	
+	RAnalBaseTypeEnum base_enum;
+	
+	char *key = sdb_fmt ("%s.%s", "enum", sanitized_name);
```

Comment:
For your info, we are trying to get rid of `sdb_fmt()` in the long run because it uses globals. We have been discussing a new design, but didn't yet agree on the concrete approach: https://github.com/radareorg/radare2/pulls?q=is%3Apr+r_strf+is%3Aunmerged

Reply:
the r_strf PR should be finished and merged

Reply:
I merged master, but I don't see any r_strf(), only r_str_newf() not sure if you meant that, but I changed to using that for now.

Reply:
then use r_str_newf() in the heap for now, its the only safe alternative. also if "enum" is always "enum", dont format it, just put it in the string